src/AppInstallerRepositoryCore/RepositorySource.cpp

```diff
@@ -417,6 +426,16 @@ namespace AppInstaller::Repository
                     details.TrustLevel = SourceTrustLevel::Trusted | SourceTrustLevel::StoreOrigin;
                     result.emplace_back(std::move(details));
                 }
+
+                if (IsMSStoreDefaultSourceEnabled())

```

Comment:
I don't want this included in the default list if it doesn't support search.  It also needs to be defined in the appropriate "get details" rather than here.

The easiest way to make it work all up would be a new bool on the details that allows it to be marked as hidden (maybe a better word to indicate that it requires direct targeting), then filter that out when creating the default composite source and returning details.  That lets it be targeted directly without making it part of the default set, as well as being able to update the details if needed.

@denelon, what would be a good word to describe a source that is not included with `winget search foo`, but can still be accessed via `winget search foo -s source`?  Hidden?  Shadow?  Explicit?

Reply:
I think I like `Restricted` the most from @denelon 's suggestions.  This obviously isn't permanent anyway, but let's use it as the starting point.